### DISCUSSION TITLE
i2c Register Auto-increment support

### DISCUSSION THREAD

**Discussion by PaulNNaish** [electronicayciencia/EasyMCP2221#10] [Q&A] [answered]
Can EasyMCP2221 use the i2c register auto increment feature? For example the PCA9685 supports this feature by setting MODE1 register bit 5 to "True".

**Comment by electronicayciencia** [comment]
Yes, with EasyMCP2221 you can set the PCA9685's MODE1 register bit 5 to "True".
Sorry... I don't get your question.

**Comment by PaulNNaish** [comment]
In auto increment you can write to successive registers by specifying only the first register index.   You do not have to specify subsequent reqister indecies.

For example of the PCA9685 if you write an i2c packet with the device address followed by register 0x16 followed by 4 Bytes followed by the stop flag you will have programmed LED 4 with the "ON" low and high bytes and the "OFF" bytes low and high bytes, in other words registers 0x16, 0x17, 0x18 & 0x19 from the same i2c packet (as long as "auto-increment" flag (register 1 bit 5) is set.  This has saved 3 seperate i2c transactions.

It is a feature of i2c.  I can't see how to do it in EASYMCP2221.

**Reply by PaulNNaish** [reply]
I think it might be "reg_bytes" but I'm struggling to get it to work.

(set to 4 to program 0x16 through to 0x19 for example)

**Comment by electronicayciencia** [comment]
Hello.

I'm sorry, I still don't get your question. But you can review this example: [examples/ledbar](https://github.com/electronicayciencia/EasyMCP2221/tree/master/examples/ledbar). It also uses the PCA9685 chip. The final result is shown in a GIF at the end of this post: [El integrado MCP2221/A](https://www.electronicayciencia.com/2023/09/integrado-mcp2221.html).

**Comment by PaulNNaish** [comment, marked as answer]
Yes, EasyMCP2221 does support automatic increment
from examples/ledbar/PCA9685.py
Line 27:  AI            = 1<<5  (AI = "AutoIncrement")

Line 115:
        if update:
            timer_on, timer_off = self.calculate_timers(duty, delay)
            v = [
                timer_on % 256,
                int(timer_on / 256),
                timer_off % 256,
                int(timer_off / 256)
                ]
            self.pca.write_register(self.LED0_ON_L + 4*led, v)

v is a 4 bytes (effectively).

It gives me enough to go on thank you.

P.

**Comment by electronicayciencia** [comment]
Perfect! Thank you.
